cmd/juju/model/defaultscommand_test.go

```diff
@@ -29,62 +29,197 @@ func (s *DefaultsCommandSuite) SetUpTest(c *gc.C) {
+			errorMatch:  `--reset accepts a single key "a" or comma delimited keys "a,b,c", received: "foo=bar"`,
+		}, {
+			// test get
+			description: "test no args inits",
+			args:        nil,
+			nilErr:      true,
+		}, {
+			description: "one key arg inits",
+			args:        []string{"one"},
+			nilErr:      true,
+		}, {
+			description: "test two key args fails",
+			args:        []string{"one", "two"},
+			errorMatch:  "can only retrieve defaults for one key or all",
+		}, {
+			description: "test multiple key args fails",
+			args:        []string{"one", "two", "three"},
+			errorMatch:  "can only retrieve defaults for one key or all",
+		}, {
+			description: "test valid region and one arg",
+			args:        []string{"dummy-region", "one"},
+			errorMatch:  "specifying a region when retrieving defaults for a setting is invalid",
+		}, {
+			description: "test valid region and no args",
+			args:        []string{"dummy-region"},
+			errorMatch:  "specifying a region when retrieving defaults is invalid",
+		}, {
+			// test cloud/region
+			description: "test invalid cloud fails",
+			args:        []string{"invalidCloud/invalidRegion", "one=two"},
+			errorMatch:  "Unknown cloud",
+		}, {
+			description: "test valid cloud with invalid region fails",
+			args:        []string{"dummy/invalidRegion", "one=two"},
+			errorMatch:  `invalid region specified: "dummy/invalidRegion"`,
+		}, {
+			description: "test no cloud with invalid region fails",
+			args:        []string{"invalidRegion", "one=two"},
+			errorMatch:  `invalid region specified: "invalidRegion"`,
+		}, {
+			description: "test valid region with set arg succeeds",
+			args:        []string{"dummy-region", "one=two"},
+			nilErr:      true,
+		}, {
+			description: "test valid region with set and reset succeeds",
+			args:        []string{"dummy-region", "one=two", "--reset", "three"},
+			nilErr:      true,
+		}, {
+			description: "test reset and set with extra key is interpereted as invalid region",
+			args:        []string{"--reset", "something,else", "invalidRegion", "is=weird"},
+			errorMatch:  `invalid region specified: "invalidRegion"`,
+		}, {
+			description: "test reset and set with valid region and extra key fails",
+			args:        []string{"--reset", "something,else", "dummy-region", "invalidkey", "is=weird"},
+			errorMatch:  "cannot set and retrieve default values simultaneously",
+		}, {
+			// test various invalid
+			description: "test too many positional args with reset",
+			args:        []string{"--reset", "a", "b", "c", "d"},
+			errorMatch:  "invalid input",
+		}, {
+			description: "test too many positional args with invalid region set",
+			args:        []string{"a", "a=b", "b", "c=d"},
+			errorMatch:  `invalid region specified: "a"`,
+		}, {
+			description: "test invalid positional args with set",
+			args:        []string{"a=b", "b", "c=d"},
+			errorMatch:  `expected "key=value", got "b"`,
```

Comment:
> Are you saying we should check for the above error and return the below error?

I'm thinking we should check for mixed get/set up-front, before we call keyvalues.Parse.